src/Widgets/PopoverWidget.vala

```diff
@@ -196,12 +233,18 @@ public class Network.Widgets.PopoverWidget : Gtk.Grid {
             if (widget_interface.is_device (device)) {
                 network_interface.remove (widget_interface);
 
+                var flowbox = (Gtk.FlowBox) widget_interface.get_ancestor (typeof (Gtk.FlowBox));
+                if (flowbox != null) {
+                    flowbox.remove (widget_interface);
+                }
```

Comment:
Something like this would be better:
```vala
unowned var parent = widget_interface.get_parent ();
if (parent is Gtk.FlowBoxChild) {
    parent.destroy ();
}
```

Reply:
Oh you're smart